Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/YamlWriter.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using YamlDotNet.RepresentationModel;
+using YamlDotNet.Serialization;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Writers
+{
+    /// <summary>
+    /// This contains the functionality to surgically manipulate a YAML blob
+    /// </summary>
+    public class YamlWriter
```

Comment:
As we discussed, both YamlWriter and JsonWriter will implement the same interface in your follow-up PR

Reply:
I have amended this PR to include the addition of `ITemplateWriter` which is implemented by `YamlWriter` and `JsonWriter`